pkg/client/cache/reflector.go

```diff
@@ -161,7 +161,8 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 			default:
 				glog.Errorf("Failed to watch %v: %v", r.expectedType, err)
 			}
-			return
+			time.Sleep(time.Second)
```

Comment:
I think I agree with Filip - my understanding of the code is that:
1. r.listerWatcher.Watch (l. 154) is and http request that is just returning a watch interface; I think it cannot return the "you're outside the watch window" error
2. r.watchHandler (l. 166) is the watch itself - this implementation is what actually can return "you're outside the watch window"

Basically, as long as you don't process the event coming from the watch in watchHandler (l. 201) [which is updating the resourceVersion in l.230], the resourceVersion is in my opinion usable (i.e. we still would like to read all event from that resourceVersion). Whether the http connection failed or not it doesn't matter.

So in my opinion this change is correct. @lavalamp - what am I missing here?